modules/nextflow/src/main/groovy/nextflow/extension/OperatorImpl.groovy

```diff
@@ -1239,4 +1240,12 @@ class OperatorImpl {
                 .getOutput()
     }
 
+    DataflowWriteChannel topic(DataflowReadChannel source, String name) {
+        if( !NF.topicChannelEnabled ) throw new MissingMethodException('topic', OperatorImpl.class, InvokerHelper.EMPTY_ARGS)
+        final op = new IntoTopicOp(source, name)
+        final target = op.apply()
+        OpCall.current.get().outputs.addAll( op.outputs )
+        return target
+    }
```

Comment:
I added a `topic` operator using `tap` as inspiration, but the problem is that it hangs. You can reproduce it by updating the e2e example as follows:
```groovy
nextflow.preview.topic = true

// ...

workflow {
  // ...
  foo | topic('foo')
}

output {
  path('data') {
    // ...
    topic 'foo', mode: 'link'
  }
}
```

@pditommaso maybe you can spot if I'm doing something wrong with the topic channel logic here

Reply:
Have a look at this 88d7eadd

Reply:
Thanks for adding the test. However the problem turned out to be that the source `foo.out` is a value channel. I guess it never sends a poison pill and the pipeline hangs.

- converting `foo.out` to a queue channel e.g. `foo.out | flatMap | topic('foo')` fixed it
- weirdly it does not happen when the topic is specified in the process definition
- it turns out `top` has the same problem, it hangs when the source is a value channel, will address separately
- `into` op seems to address it by sending a poison pill when necessary

So I guess we need to either (1) ensure the source channel sends a poison pill or (2) change the topic internal behavior to handle this case

Tbh I'm not sure why the topic behaves this way since it's just a mix op under the hood. I will copy the `into` op approach for now